tutorials/en/vectara-advanced-app-tutorial-showcase-the-creation-of-vectara-app-in-legal-or-customer-support-use-case.mdx

```diff
@@ -0,0 +1,671 @@
+
+### 🛠 Fundamental Workings and Workflow
+
+The heartbeat of Vectara is its pure neural search platform enriched with production-ready natural language processing. The workflow is simple, yet powerful:
+
+1. **Data Ingestion**: Ingest your data into Vectara's corpus using the Indexing API.
+2. **Data Indexing**: The ingested data is indexed, storing vector encodings optimized for low latency and high recall.
+3. **Query Execution**: Utilize the Search API to run queries against the indexed data, retrieving highly relevant information swiftly.
+
+The beauty of Vectara lies in its API-addressable platform, which is a canvas for developers to paint their search solutions, embedding them within their applications.
+
+---
+### 🎨 Dive into Vectara's Console
+
+To truly grasp the potential of Vectara, let's delve into its console which is the epicenter of managing your search ecosystem:
+
+- **Creating Corpora**: Start by creating a corpus, a sanctuary for your data awaiting to be queried. The process is straightforward - name your corpus, describe it, select an embedding model, specify filter attributes, and voila, your corpus is ready to be fed with data.
+- **API Access Management**: Vectara grants you the wand to manage API access. Create and manage API keys and app clients with ease. With the necessary permissions, an `API access` tab unveils in the sidebar, guiding you to create API keys and app clients. It's your doorway to interact with Vectara's treasure trove of search capabilities.
+- **Team Collaboration**: Invite your team to the Vectara console, assign specific roles, and foster a collaborative environment to build and refine your search solutions.
+- **Billing Management**: Keep a tab on your account usage and manage billing details ensuring uninterrupted service as you sail through the Vectara ecosystem.
+
+
+In this section, we've skimmed the surface of Vectara's offerings. As we delve deeper into our chosen use case in the next section, the utility and power of Vectara will unfold further, painting a clearer picture of how it can be harnessed for Customer Support applications.
+
+---
+## Our Quest: Orchestrating a Customer Support Maestro 🎵
+
+Embarking on the expedition to create a Customer Support solution requires a sturdy vessel, and Vectara proves to be an admirable choice for this voyage. Unlike the conventional techniques that revolve around wrapping OpenAI APIs without delving deeper into the core concepts, Vectara emerges as a beacon of innovation especially in the wake of GPT-4's multi-modal capabilities which sent ripples through the RAG-based companies recently. Vectara simplifies the creation of QA and conversational based projects, extending its arms to developers with its pre-built models and custom workflows. This eradicates the superficial approach often taken by budding developers who get entangled in ready-made wrappers like Langchain, without a thorough understanding of the underlying principles and optimizations.
```

Comment:
I don't understand this part. 
- "especially in the wake of GPT-4's multi-modal capabilities which sent ripples through the RAG-based companies recently" - why is the multi-modal capabilities related to RAG? and what do you mean by "RAG-based companies"?

Reply:
Also this sentence "This eradicates the superficial approach often taken by budding developers who get entangled in ready-made wrappers like Langchain, without a thorough understanding of the underlying principles and optimizations." - I understand what you are trying to say, but I would suggest simplifying the question and making the value proposition clearer. For example "This eradicates the superficial approach often taken by budding developers who get start with ready-made wrappers like LlamaIndex, only to realize later the complexities involved in building chatbot and conversational AI systems."
(Using LlamaIndex instead of LangChain - since they are parttially involved in the hackathon)

Reply:
I understand how the term 'RAG-based companies' and the connection to GPT-4's multi-modal capabilities might be unclear. Allow me to elaborate on that.

Recently, there has been a surge in startups and companies that have leveraged RAG to create applications centered around conversational AI, essentially building wrappers around tools like ChatGPT to facilitate document-based chats.

However, Vectara stands out distinctly in this landscape. Unlike the aforementioned startups that predominantly rely on RAG, Vectara has developed its own comprehensive suite of tools and capabilities. While we do utilize GPT for certain functionalities like summarization, Vectara has gone a step further to incorporate better  indexing and other crucial features, creating a more robust and integrated solution.

With the advent of GPT-4, we witnessed an introduction of multi-modal capabilities, allowing the model to understand and generate content across various formats, not limited to text. This development is pivotal as it challenges the existing RAG-based applications, pushing them to evolve and adapt, lest they risk becoming obsolete. It signifies a shift towards more versatile and all-encompassing AI tools.